mqtt.py

```diff
@@ -31,196 +32,204 @@ def __init__(self, config, data):
-                    info.ParseFromString(mp.decoded.payload)
-                    out = json.loads(MessageToJson(info, preserving_proto_field_name=True, ensure_ascii=False, indent=2, sort_keys=True, use_integers_for_enums=True, always_print_fields_with_no_presence=True))
-                    outs["type"] = "neighborinfo"
-                    outs["payload"] = out
-                    print(f"Decoded protobuf message: neighborinfo: {outs}")
-                    self.handle_neighborinfo(outs)
-
-                elif mp.decoded.portnum == portnums_pb2.NODEINFO_APP:
-                    info = mesh_pb2.User()
-                    info.ParseFromString(mp.decoded.payload)
-                    out = json.loads(MessageToJson(info, preserving_proto_field_name=True, ensure_ascii=False, indent=2, sort_keys=True, use_integers_for_enums=True))
-                    out["id"] = out['id'].replace('!', '')
-                    outs["type"] = "nodeinfo"
-                    outs["payload"] = out
-                    print(f"Decoded protobuf message: nodeinfo: {outs}")
-                    self.handle_nodeinfo(outs)
-
-                elif mp.decoded.portnum == portnums_pb2.ROUTING_APP:
-                    data = mesh_pb2.Routing()
-                    data.ParseFromString(mp.decoded.payload)
-                    out = json.loads(MessageToJson(data, preserving_proto_field_name=True, ensure_ascii=False, indent=2, sort_keys=True, use_integers_for_enums=True))
-                    outs["type"] = "routing"
-                    outs["payload"] = out
-                    print(f"Decoded protobuf message: routing: {outs}")
-                    # self.handle_routing(outs)
-
-                elif mp.decoded.portnum == portnums_pb2.TRACEROUTE_APP:
-                    route = mesh_pb2.RouteDiscovery()
-                    route.ParseFromString(mp.decoded.payload)
-                    out = json.loads(MessageToJson(route, preserving_proto_field_name=True, ensure_ascii=False, indent=2, sort_keys=True, use_integers_for_enums=True, always_print_fields_with_no_presence=True))
-                    if 'route' in out:
+                    print(f"*** Decryption failed: {str(e)}")
+                    return
+                is_encrypted = True
+
+            outs['rssi'] = mp.rx_rssi
+            outs['snr'] = mp.rx_snr
+            outs['timestamp'] = mp.rx_time
+
+            # TODO: Need to handle this
+            # self.handle_log(mp.decoded)
+
+            if mp.decoded.portnum == portnums_pb2.TEXT_MESSAGE_APP:
+                text = mp.decoded.payload.decode("utf-8")
+                payload = { "text": text }
+                outs["type"] = "text"
+                outs["payload"] = payload
+                print(f"Decoded protobuf message: text: {outs}")
+                self.handle_text(outs)
+
+            elif mp.decoded.portnum == portnums_pb2.MAP_REPORT_APP:
+                report = mesh_pb2.Position()
+                report.ParseFromString(mp.decoded.payload)
```

Comment:
FYI protos have a class method `FromString` so you could do `report = mesh_pb2.Position.FromString(mp.decoded.payload)`

Reply:
Oooh nice, I'll have to move to that.

Reply:
Alright, pushed up with those in place. Thanks for the tip.